app/src/main/java/fr/free/nrw/commons/media/MediaDetailFragment.java

```diff
@@ -436,8 +436,18 @@ private void enableDeleteButton(boolean visibility) {
     private void rebuildCatList() {
         categoryContainer.removeAllViews();
         // @fixme add the category items
-        for (String cat : categoryNames) {
-            View catLabel = buildCatLabel(cat, categoryContainer);
+
+        //As per issue #1826, some categories come suffixed with strings prefixed with |. As per the discussion
+        //that was meant for alphabetical sorting of the categories and can be safely removed.
+        for(int i=0;i<categoryNames.size();i++){
+            String categoryName = categoryNames.get(i);
+            //Removed everything after '|'
+            if (categoryName.indexOf('|') != -1) {
```

Comment:
I see you are executing .indexOf('|') twice in the cases where it is found. It would be more efficient to store the index value found so it can be reused and always executed only once.

Reply:
I agree @albendz

Reply:
Agreed, should have been like that..have done the changes :)